### PR TITLE
docs: Add a checklist for new modules

### PR DESCRIPTION
**Pull request by chipbuster**
#### Description
Adds a module checklist in CONTRIBUTING.md to make it easier to remember (for both authors and reviewers)
what steps need to be taken when merging a new module.

#### Motivation and Context
It's really hard to remember everything that needs to be done sometimes.

This is very much a draft and I would appreciate any suggestions for how to change the wording/structure.

#### How Has This Been Tested?
N/A, but would appreciate to see if there's anything I missed.

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -225,3 +225,25 @@ This is our preferred process for opening a PR on GitHub:
...
+      appropriate--this is a bare minimum).
+- [ ] Add the variable to the appropriate location in the "Default Prompt 
+      Format" section of the documentation
+- [ ] Add an appropriate choice of options to each preset in `docs/presets/README.md`
+- [ ] Create configs structs/traits in `src/configs/<module>.rs` and add to the
+      appropriate locations in:
+  - [ ] `PROMPT_ORDER` (`src/configs/starship_root.rs`)
+  - [ ] `FullConfig` and the `Default` impl (`src/configs/mod.rs`)
+  - [ ] `ALL_MODULES` (`src/module.rs`)
+
```
Comment (davidkna) [inline CONTRIBUTING.md:247]: I think `modules/mod.rs` with `handle()` and `description()` should also be mentioned in the list.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -225,3 +225,25 @@ This is our preferred process for opening a PR on GitHub:
...
+Finally, you should make sure to write your module's code in `src/modules`
+and add any commands that need to be mocked when testing in `src/utils.rs`.
```
Comment (davidkna) [inline CONTRIBUTING.md:249]: For mocking, I think the dynamic mocking with `ModuleRenderer::cmd` should also be mentioned.

Comment (chipbuster) [reply CONTRIBUTING.md:249]: I'm not 100% sure how this should be worded. I've pushed a change, but I don't think it captures the idea very well.

**Review by matchai — APPROVED** [review]
Love the addition! ❤️ 
Looks good to me.

**Pull request merged**